app/helpers/quadicon_helper.rb

```diff
@@ -641,6 +699,15 @@ def render_single_quad_quadicon(item, options)
     output.collect(&:html_safe).join('').html_safe
   end
 
+  def img_for_health_state(item)
```

Comment:
I'm impressive with all ManageIQ archtecture, then I think that refactoring which you proposed can be think after this PR be merged, because this feature it's taking more time than supposed to.